src/components/MenuButton/index.tsx

```diff
@@ -97,6 +97,7 @@ export const MenuItem = styled<React.FC<MenuItemProps>>(ReachMenuItem)`
 
 export interface MenuLinkProps extends ReachMenuLinkProps {
   href: string | undefined
+  as: string | undefined
```

Comment:
Isn't it better to make it `nullable` instead of using `undefined`? Same goes for `href`